src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+
+            // Handler creates session of type T for specific port on connection.
+            handler<auth_session> auth(io_context, 54231);
+            handler<view_session> view(io_context, 54001);
+            handler<data_session> data(io_context, 54230);
+            asio::steady_timer    cleanup_callback(io_context, std::chrono::minutes(15));
+
+            cleanup_callback.async_wait(std::bind(&ConnectServer::periodicCleanup, this, std::placeholders::_1, &cleanup_callback));
+
+            // NOTE: io_context.run() takes over and blocks this thread. Anything after this point will only fire
+            // if io_context finishes!
+            ShowInfo("starting io_context");
+
+            io_context.run();
+        }
+        catch (std::exception& e)
+        {
+            ShowError(e.what());
+        }
+    }
+
+    ~ConnectServer() override
+    {
+        // Everything should be handled with RAII
+    }
+
+    // TODO: Currently never called. Need io_context asio::steady_timer callback with taskmgr to control timing?
+    void Tick() override
+    {
+        Application::Tick();
+
+        // Connect Server specific things
+    }
+
+    // This cleanup function is to periodically poll for auth sessions that were successful but xiloader failed to actually launch FFXI
+    // When this happens, the data/view socket are never opened and will never be cleaned up normally.
+    // Auth is closed before any other sessions are open, so the data/view cleanups aren't sufficient
+    void periodicCleanup(const asio::error_code& error, asio::steady_timer* timer)
+    {
+        if (!error)
+        {
+            auto ipAddrIterator = authenticatedSessions_.begin();
+            while (ipAddrIterator != authenticatedSessions_.end())
+            {
+                auto sessionIterator = ipAddrIterator->second.begin();
+                while (sessionIterator != ipAddrIterator->second.end())
+                {
+                    session_t& session = sessionIterator->second;
+
+                    // If it's been 15 minutes, erase it from the session list
+                    if (!session.data_session &&
+                        !session.view_session &&
+                        (server_clock::now() - session.authorizedTime) > std::chrono::minutes(15))
+                    {
+                        sessionIterator = ipAddrIterator->second.erase(sessionIterator);
+                    }
+                    else
+                    {
+                        sessionIterator++;
+                    }
+                }
+
+                // If this map entry is empty, clean it up
+                if (ipAddrIterator->second.size() == 0)
+                {
+                    ipAddrIterator = authenticatedSessions_.erase(ipAddrIterator);
+                }
+                else
+                {
+                    ipAddrIterator++;
+                }
+            }
+
+            if (Application::IsRunning())
```

Comment:
Given how long this file is, consider how you might be able to break it up into smaller files (given that there still needs to be a .h/.cpp cleanup already)

Reply:
Broke out everything that made sense into .h files